x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -111,24 +113,86 @@ export class RuleDataClient implements IRuleDataClient {
+      body: {
+        index_patterns: [`${primaryNamespacedAlias}-*`],
+        composed_of: [...this.options.componentTemplateNames],
+        template: {
+          aliases:
+            secondaryNamespacedAlias != null
+              ? {
+                  [secondaryNamespacedAlias]: {
+                    is_write_index: false,
+                  },
+                }
+              : undefined,
+          settings: {
+            'index.lifecycle': {
+              name: DEFAULT_ILM_POLICY_ID,
+              // TODO: fix the types in the ES package, they don't include rollover_alias???
+              // @ts-expect-error
+              rollover_alias: primaryNamespacedAlias,
+            },
+          },
+        },
+        _meta: {
+          namespace,
+        },
+        // By setting the priority to namespace.length, we ensure that if one namespace is a prefix of another namespace
+        // then newly created indices will use the matching template with the *longest* namespace
+        priority: namespace?.length,
+      },
+    };
+  }
+
   async createWriteTargetIfNeeded({ namespace }: { namespace?: string }) {
-    const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    const primaryNamespacedAlias = getNamespacedAlias({ alias: this.options.alias, namespace });
 
     const clusterClient = await this.getClusterClient();
 
     const { body: aliasExists } = await clusterClient.indices.existsAlias({
-      name: alias,
+      name: primaryNamespacedAlias,
     });
 
-    const concreteIndexName = `${alias}-000001`;
+    const concreteIndexName = `${primaryNamespacedAlias}-000001`;
 
     if (!aliasExists) {
+      const secondaryNamespacedAlias =
+        this.options.secondaryAlias != null
+          ? getNamespacedAlias({ alias: this.options.secondaryAlias, namespace })
+          : undefined;
+      const template = this.createNamespacedIndexTemplate({
+        primaryNamespacedAlias,
+        secondaryNamespacedAlias,
+        namespace,
+      });
+      // TODO: need a way to update this template if/when we decide to make changes to the
+      // built in index template. Probably do it as part of updateIndexMappingsForAsset?
+      // (Before upgrading any indices, find and upgrade all namespaced index templates - component templates
+      // will already have been upgraded by solutions or rule registry, in the case of technical/ECS templates)
+      // With the current structure, it's tricky because the index template creation
+      // depends on both the namespace and secondary alias, both of which are not currently available
+      // to updateIndexMappingsForAsset. We can make the secondary alias available since
+      // it's known at plugin startup time, but
+      // the namespace values can really only come from the existing templates that we're trying to update
+      // - maybe we want to store the namespace as a _meta field on the index template for easy retrieval
```

Comment:
Yeah, we will need to somehow retrieve the namespace from existing index templates, I think it's a good idea to store it in `_meta` 👍  Thanks for this idea!